src/flakeModule.nix

```
@@ -0,0 +1,138 @@
+    inherit
+      (types.either
+        (types.listOf types.nonEmptyStr)
+        (types.listOf (types.listOf types.nonEmptyStr)))
+      check
+      ;
+  };
+
+  cellBlocksType = mkOptionType {
+    name = "cell block";
+    description = "defines the global set of output/target types of the project";
+    descriptionClass = "noun";
+    inherit (validators.CellBlocks) check;
+  };
+
+  opt = options.std;
+  cfg = config.std;
+in {
+  _file = ./flakeModule.nix;
+  options = {
+    std = {
+      grow = mkOption {
+        description = "Orderly 'grow' a project from Standard Cells & Cell Blocks";
+        type = types.submodule {
+          options = {
+            cellsFrom = mkOption {
+              description = "Where Standard discovers Cells from.";
+              type = types.path;
+            };
+            cellBlocks = mkOption {
+              description = "Declaration of all Cell Blocks used in the project.";
+              type = cellBlocksType;
+            };
+            nixpkgsConfig = mkOption {
+              description = "Nixpkgs configuration applied to `inputs.nixpkgs` (if that input exists).";
+              type = types.attrs;
+            };
+          };
+        };
+      };
+      pick = mkOption {
+        description = "Pick Standard outputs. Like 'harvest' but remove the system for outputs that are system agnostic.";
+        type = types.attrsOf harvesterType;
+      };
+      winnowIf = mkOption {
+        description = "predicate for winnow";
+        type = types.attrsOf (types.functionTo (types.functionTo types.bool));
+      };
+      winnow = mkOption {
+        description = "Winnow Standard outputs. Like 'harvest', but with a filter predicate 'winnowIf'.";
+        type = types.attrsOf harvesterType;
+      };
+      harvest = mkOption {
+        description = "Harvest Standard outputs into a Nix-CLI-compatible form (a.k.a. the 'official' flake schema).";
+        type = types.attrsOf harvesterType;
+      };
+    };
+  };
+  config = {
+    flake = let
+      grown = grow (cfg.grow
+        // {
+          inherit (self) inputs;
+          inherit (config) systems;
+          # access them explicitly to trigger a module system error if not defined
+          inherit (cfg.grow) cellsFrom cellBlocks;
+        });
+      picked = mapAttrs (_: v: pick grown v) cfg.pick;
+      harvested = mapAttrs (_: v: harvest grown v) cfg.harvest;
+      winnowed = zipAttrsWith (n: v: winnow (head v) grown (head (tail v))) [cfg.winnowIf cfg.winnow];
+    in
+      lib.foldl' lib.recursiveUpdate {} (
+        [grown]
+        ++ (lib.optionals opt.pick.isDefined [picked])
+        ++ (lib.optionals (opt.winnow.isDefined && opt.winnowIf.isDefined) [winnowed])
+        ++ (lib.optionals opt.harvest.isDefined [harvested])
+      );
+
+    # exposes the raw scheme of the std layout inside flake-parts
+    perInput = system: flake: lib.optionalAttrs (lib.hasAttrs system flake) flake.${system};
```

I'd recommend namespacing this, to avoid collisions. `perInput` is applied to _all_ inputs to form `inputs'`, so setting arbitrary attributes could lead to nasty interactions. I'll improve the `perInputs` description.

```suggestion
    perInput = system: flake: { std = flake.${system} or {}; };
```

---

Judging from the :+1: I think you forgot this one.